### PR TITLE
Some checks for file open and file has keyword.

### PR DESCRIPTION
**Pull request by joakim-hove**
Trying to chase the current Travis + Jenkins failure in the `compare_restart_files` test. Unfortunately it works on my machine&trade; - so a bit difficult to debug. The current PR has some solidifying changes which make sense anyway.

### CODE REVIEW

File: tests/test_restart.cpp [hunk 1]
```diff
@@ -52,23 +54,39 @@ BOOST_AUTO_TEST_CASE(CompareRestartFileResults)
...
 
-    for (auto key : {"PRESSURE", "SWAT", "SGAS", "RS", "RV"}) {
-        ecl_kw_type * kw_1 =  ecl_file_iget_named_kw( file1 , key, 0);
-        ecl_kw_type * kw_2 =  ecl_file_iget_named_kw( file2 , key, 0);
 
-        bool numeric_equal = ecl_kw_numeric_equal(kw_1, kw_2, abs_diff , relative_diffs[key]);
-        if (numeric_equal) {
-            std::cout << " Restart results for " << key << " compared ok" << std::endl;
+    if (!file1) {
+        std::cerr << "Failed to open restart file: " << filename1.c_str() << std::endl;
```
Comment (bska) [inline tests/test_restart.cpp]: Just curious about one tiny detail: Is there a reason you're using `.c_str()` here?

**Comment by pgdr** [comment]
Travis failed with the same compare_restart_file problem as before.

          Start 18: compare_restart_files
    18/18 Test #18: compare_restart_files ............***Failed    0.01 sec
    Running 1 test case...
    Failed to open restart file: SPE1CASE2.UNRST
    94% tests passed, 1 tests failed out of 18
    Total Test time (real) =  15.63 sec
    The following tests FAILED:
    	 18 - compare_restart_files (Failed)
    Errors while running CTest

File: tests/test_restart.cpp (hunk 1 above)
Comment (joakim-hove) [inline tests/test_restart.cpp]: > Is there a reason you're using .c_str() here?

Ehhh no - it was a mistake.

**Comment by joakim-hove** [comment]
>  with the same compare_restart_file problem as before.

Well - now we have an error message: "`Failed to open restart file: SPE1CASE2.UNRST`"

**Comment by joakim-hove** [comment]
This PR is quite trivial - unless someone beats me to it I will self-merge it tomorrow.

File: tests/test_restart.cpp [hunk 2]
```diff
@@ -52,23 +54,39 @@ BOOST_AUTO_TEST_CASE(CompareRestartFileResults)
...
     ecl_file_type*  file2 =  ecl_file_open_rstblock_report_step( filename2.c_str() , last_report_step, 1);
 
-    for (auto key : {"PRESSURE", "SWAT", "SGAS", "RS", "RV"}) {
-        ecl_kw_type * kw_1 =  ecl_file_iget_named_kw( file1 , key, 0);
-        ecl_kw_type * kw_2 =  ecl_file_iget_named_kw( file2 , key, 0);
 
-        bool numeric_equal = ecl_kw_numeric_equal(kw_1, kw_2, abs_diff , relative_diffs[key]);
-        if (numeric_equal) {
-            std::cout << " Restart results for " << key << " compared ok" << std::endl;
+    if (!file1) {
```
Comment (atgeirr) [inline tests/test_restart.cpp]: These should use the BOOST_REQUIRE macro (test will terminate after unsuccessful BOOST_REQUIRE unlike BOOST_CHECK).

File: tests/test_restart.cpp [hunk 3]
```diff
@@ -52,23 +54,39 @@ BOOST_AUTO_TEST_CASE(CompareRestartFileResults)
...
+        std::exit(1);
+    }
+
+    if (!file2) {
+        std::cerr << "Failed to open restart file: " << filename2 << std::endl;
+        std::exit(1);
+    }
+
+    for (const char * key : {"PRESSURE", "SWAT", "SGAS", "RS", "RV"}) {
+        if ((ecl_file_has_kw( file1 , key)) && (ecl_file_has_kw( file2 , key))) {
```
Comment (atgeirr) [inline tests/test_restart.cpp]: This should use BOOST_CHECK, letting the tests for other keywords continue even if a keyword was not found. For example:
```
const bool found_keywords = ecl_file_has_kw( file1 , key) && ecl_file_has_kw( file2 , key);
BOOST_CHECK(found_keywords);
if (found_keywords) {
    ...
}
```

File: tests/test_restart.cpp (hunk 2 above)
Comment (joakim-hove) [inline tests/test_restart.cpp]: OK - good idea.

File: tests/test_restart.cpp (hunk 3 above)
Comment (joakim-hove) [inline tests/test_restart.cpp]: > This should use BOOST_CHECK,

Well - if one keyword is missing the files are broken beyond repair anyway; do not see very much point in continuing.

Comment (atgeirr) [inline tests/test_restart.cpp]: I was thinking perhaps at some point one keyword is written the wrong way and the rest could be ok, but you are right that it seems very unlikely. Please ignore the previous request.

Comment (atgeirr) [inline tests/test_restart.cpp]: (Sorry I forgot) That of course points towards using `BOOST_REQUIRE`.

**Pull request merged**